engines/ospray/ispc/geometry/SDFBeziers.ispc

```diff
@@ -0,0 +1,444 @@
+
+uniform box3fa bezierBounds(dataPtr_t geo_, dataPtr_t bezier_)
+{
+#define USE_TIGHT_BOUNDS
+#ifndef USE_TIGHT_BOUNDS
+    const Bezier_ptr bc = (const Bezier_ptr)bezier_;
+    const uniform vec3f mi = min(min(bc->p0 - bc->r0, bc->p1 - bc->r1),
+                                 min(bc->c0 - bc->r0, bc->c1 - bc->r1));
+    const uniform vec3f ma = max(max(bc->p0 + bc->r0, bc->p1 + bc->r1),
+                                 max(bc->c0 + bc->r0, bc->c1 + bc->r1));
+    return make_box3fa(mi, ma);
+#else
+    const Bezier_ptr bc = (const Bezier_ptr)bezier_;
+
+    uniform float r = max(bc->r0, bc->r1);
+
+    // tight bounding box
+    uniform vec3f mi = min(bc->p0, bc->p1);
+    uniform vec3f ma = max(bc->p0, bc->p1);
+
+    const uniform vec3f c = -1.0f * bc->p0 + bc->c0;
+    const uniform vec3f b = 1.0f * bc->p0 - 2 * bc->c0 + bc->c1;
+    const uniform vec3f a = -1.0f * bc->p0 + 3 * bc->c0 - 3 * bc->c1 + bc->p1;
+
+    uniform vec3f h = b * b - a * c;
+
+    if (h.x > 0.0f)
+    {
+        h.x = sqrt(h.x);
+        uniform float t = (-b.x - h.x) / a.x;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.x, bc->c0.x, bc->p1.x, bc->c1.x);
+            mi.x = min(mi.x, q);
+            ma.x = max(ma.x, q);
+        }
+
+        t = (-b.x + h.x) / a.x;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.x, bc->c0.x, bc->p1.x, bc->c1.x);
+            mi.x = min(mi.x, q);
+            ma.x = max(ma.x, q);
+        }
+    }
+
+    if (h.y > 0.0f)
+    {
+        h.y = sqrt(h.y);
+        uniform float t = (-b.y - h.y) / a.y;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.y, bc->c0.y, bc->p1.y, bc->c1.y);
+            mi.y = min(mi.y, q);
+            ma.y = max(ma.y, q);
+        }
+        t = (-b.y + h.y) / a.y;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.y, bc->c0.y, bc->p1.y, bc->c1.y);
+            mi.y = min(mi.y, q);
+            ma.y = max(ma.y, q);
+        }
+    }
+
+    if (h.z > 0.0f)
+    {
+        h.z = sqrt(h.z);
+        uniform float t = (-b.z - h.z) / a.z;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.z, bc->c0.z, bc->p1.z, bc->c1.z);
+            mi.z = min(mi.z, q);
+            ma.z = max(ma.z, q);
+        }
+        t = (-b.z + h.z) / a.z;
+        if (!isnan(t) && t > 0.0f && t < 1.0f)
+        {
+            uniform float q = bezier(t, bc->p0.z, bc->c0.z, bc->p1.z, bc->c1.z);
+            mi.z = min(mi.z, q);
+            ma.z = max(ma.z, q);
+        }
+    }
+
+    return make_box3fa(mi - r, ma + r);
+#endif
+}
+//////////////////////////////////////////////////////////////////////
+
+#define MIN_SEGMENT_COUNT 3
+#define MAX_SEGMENT_COUNT 9
+
+int screenAwareSegmentCount(const Bezier_ptr bc, const vec4f& params)
```

Comment:
In the future I think we should have a command-line parameter for specifying the quality of the bezier curves, if it is hard to find an automatic value.